opentelemetry-otlp/src/lib.rs

```diff
@@ -183,7 +183,7 @@
 //!         .with_resource(Resource::new(vec![KeyValue::new("service.name", "example")]))
 //!         .with_period(Duration::from_secs(3))
 //!         .with_timeout(Duration::from_secs(10))
-//!         .with_temporality_selector(DefaultTemporalitySelector::new())
+//!         .with_temporality(Temporality::default())
```

Comment:
Following the same thing as https://github.com/open-telemetry/opentelemetry-rust/pull/2217#discussion_r1805266835, we could just remove this.